pkg/proto/types.go

```diff
@@ -3138,6 +3169,30 @@ func (e *DataEntries) UnmarshalJSON(data []byte) error {
 	return nil
 }
 
+// Equal compares two DataEntries slices and returns true if they are equal.
+func (e DataEntries) Equal(other DataEntries) (bool, error) {
```

Comment:
It's necessary to say that this function mutates passed argument, i.e. elements order in the slice.